bmap/src/lib.rs

```diff
@@ -4,6 +4,7 @@ mod discarder;
 pub use crate::discarder::*;
 use sha2::{Digest, Sha256};
 use thiserror::Error;
+use nix::unistd::ftruncate;
```

Comment:
suggest to split this into two commits:
1. to move the ftruncate into the `bmap::copy` function,
2. to call ftruncate only on a file :-)